server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -170,7 +170,17 @@ protected TransportReplicationAction(Settings settings, String actionName, Trans
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
         assert request.shardId() != null : "request shardId must be set";
-        runReroutePhase(task, request, listener, true);
+        request.incRef();
```

Comment:
Could we maybe move this into `runReroutePhase`? Seems like we do the same in both call sites.

Reply:
++ dried this up now in both spots :)